### PR TITLE
Add cart tests

### PR DESCRIPTION
**Pull request by AntonoffEvgeniy**
Add cart tests

### CODE REVIEW

File: ShopAppTests/App/Cart/Views/Cells/TableView/CartTableViewCellSpec.swift [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+            
+            quantityLabel = self.findView(withAccessibilityLabel: "quantityLabel", in: cell) as! UILabel
+            quantityTextFieldView = self.findView(withAccessibilityLabel: "quantityTextFieldView", in: cell) as! QuantityTextFieldView
+            variantImageView = self.findView(withAccessibilityLabel: "variantImageView", in: cell) as! UIImageView
+            variantTitleLabel = self.findView(withAccessibilityLabel: "variantTitleLabel", in: cell) as! UILabel
+            totalPriceLabel = self.findView(withAccessibilityLabel: "totalPriceLabel", in: cell) as! UILabel
+            pricePerOneItemLabel = self.findView(withAccessibilityLabel: "pricePerOneItemLabel", in: cell) as! UILabel
+        }
+        
+        describe("when cell initialized") {
```
Comment (radyslavkrechet) [inline ShopAppTests/App/Cart/Views/Cells/TableView/CartTableViewCellSpec.swift]: Also test
`pricePerOneItemLabel.isHidden = true`
that setup in `didSet` of `pricePerOneItemLabel` outlet.

File: ShopAppTests/App/Cart/Views/Cells/TableView/CartTableViewCellSpec.swift [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+            var cartProduct: CartProduct!
+            
+            context("if cart product nil") {
+                beforeEach {
+                    cell.configure(with: nil)
+                }
+                
+                it("should have correct image") {
+                    expect(variantImageView.image).toNot(beNil())
+                    expect(variantImageView.image) == #imageLiteral(resourceName: "placeholder")
```
Comment (radyslavkrechet) [inline ShopAppTests/App/Cart/Views/Cells/TableView/CartTableViewCellSpec.swift]: I think this and next `expect` should be in spec of `UIImageView` extension.

File: ShopAppTests/App/Cart/Views/Cells/TableView/CartTableViewCellSpec.swift [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+            
+            it("should have correct quantity tet field delegate") {
+                expect(quantityTextFieldView.delegate) === cell
+            }
+        }
+        
+        describe("when cell configured") {
+            var cartProduct: CartProduct!
+            
+            context("if cart product nil") {
```
Comment (radyslavkrechet) [inline ShopAppTests/App/Cart/Views/Cells/TableView/CartTableViewCellSpec.swift]: Also test that `cartProduct` is nil in this and other contexts because you setup it in `configure` method.

File: ShopAppTests/App/Cart/ViewModels/CartViewModelSpec.swift [hunk 4]
```diff
@@ -0,0 +1,272 @@
...
+                let productVariantSecond = ProductVariant()
+                productVariantSecond.price = Decimal(floatLiteral: 15)
+                cartProductSecond.productVariant = productVariantSecond
+            }
+            
+            it("should calculate total price") {
+                viewModel.data.value = [cartProductFirst, cartProductSecond]
+                let totalPrice = viewModel.calculateTotalPrice()
+                
+                expect(totalPrice) == 5 * 10 + 10 * 15
```
Comment (radyslavkrechet) [inline ShopAppTests/App/Cart/ViewModels/CartViewModelSpec.swift]: I think we can use concrete value for test and left math operations as comment, something like this:
 `expect(totalPrice) == 200 // 5 * 10 + 10 * 15`

File: ShopAppTests/App/Cart/Views/Cells/TableView/CartTableViewCellSpec.swift (hunk 3 above)
Comment (AntonoffEvgeniy) [reply ShopAppTests/App/Cart/Views/Cells/TableView/CartTableViewCellSpec.swift]: This property has private protection level in CartTableViewCell class

Comment (radyslavkrechet) [reply ShopAppTests/App/Cart/Views/Cells/TableView/CartTableViewCellSpec.swift]: Okay, I've found that you had changed it in this PR.

File: ShopApp/App/Modules/Cart/Views/Headers/CartHeaderView/CartHeaderView.swift [hunk 5]
```diff
@@ -8,7 +8,7 @@
 
 import UIKit
 
-let kCartHeaderViewHeader: CGFloat = 76
+let kCartHeaderViewHeight: CGFloat = 76
```
Comment (plzen) [inline ShopApp/App/Modules/Cart/Views/Headers/CartHeaderView/CartHeaderView.swift]: Move constant to class

File: ShopAppTests/App/Cart/Controllers/CartViewControllerSpec.swift [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+//
+
+import Nimble
+import Quick
+import ShopApp_Gateway
+import SwipeCellKit
+
+@testable import ShopApp
+
+class CartViewControllerSpec: QuickSpec {
```
Comment (plzen) [inline ShopAppTests/App/Cart/Controllers/CartViewControllerSpec.swift]: Add tests for SwipeTableViewCellDelegate

File: ShopAppTests/App/Cart/Views/Cells/TableView/CartTableViewCellSpec.swift [hunk 7]
```diff
@@ -0,0 +1,219 @@
...
+//  Copyright © 2018 RubyGarage. All rights reserved.
+//
+
+import Nimble
+import Quick
+import ShopApp_Gateway
+
+@testable import ShopApp
+
+class CartTableViewCellSpec: QuickSpec {
```
Comment (plzen) [inline ShopAppTests/App/Cart/Views/Cells/TableView/CartTableViewCellSpec.swift]: Add tests for populatePricePerOne in line 102

Comment (AntonoffEvgeniy) [reply ShopAppTests/App/Cart/Views/Cells/TableView/CartTableViewCellSpec.swift]: This method accepts non optional property, but return optional. I haven't any ideas how to test this line
